core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -98,6 +98,8 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
   private val reviveThread =
     ThreadUtils.newDaemonSingleThreadScheduledExecutor("driver-revive-thread")
 
+  private var driverEndpointStopped = false
```

Comment:
shouldn't this be inside  class DriverEndpoint?

Reply:
If we keep the `driverEndpointStopped` inside `DriverEndpoint`, need to have another var for `DriverEndpoint` ref from `rpcEnv.setupEndpoint(ENDPOINT_NAME, createDriverEndpoint(properties))` to access the `DriverEndpoint.driverEndpointStopped`.